api/base/exceptions.py

```diff
@@ -1,5 +1,36 @@
-from rest_framework.exceptions import APIException
+
 from rest_framework import status
+from rest_framework.exceptions import APIException
+
+
+def json_api_exception_handler(exc, context):
+    """ Custom exception handler that returns errors object as an array """
+
+    # Import inside method to avoid errors when the OSF is loaded without Django
+    from rest_framework.views import exception_handler
+    response = exception_handler(exc, context)
+
+    # Error objects may have the following members. Title removed to avoid clash with node "title" errors.
+    top_level_error_keys = ['id', 'links', 'status', 'code', 'detail', 'source', 'meta']
+    errors = []
+
+    if response:
+        message = response.data
+        if isinstance(message, dict):
+            for key, value in message.iteritems():
+                if key in top_level_error_keys:
+                    errors.append({key: value})
+                else:
+                    errors.append({'detail': {key: value}})
```

Comment:
That's a good idea, I like using meta to get the field name.  Another instance where this branch is hit is when trying to update a user without supplying fullname:

![screen shot 2015-09-02 at 11 49 11 am](https://cloud.githubusercontent.com/assets/9755598/9636278/967b32a2-5168-11e5-91e9-4906c20050db.png)